src/Microsoft.TestPlatform.Utilities/CodeCoverageDataAttachmentsHandler.cs

```diff
@@ -95,6 +113,20 @@ private async Task MergeCodeCoverageFilesAsync(IList<string> files, Cancellation
                     cancellationToken).ConfigureAwait(false);
 
             coverageUtility.WriteCoverageFile(files[0], coverageData);
+
+            foreach(var file in files.Skip(1))
+            {
+                try
+                {
+                    File.Delete(file);
```

Comment:
Deleting original files after merge?,

Reply:
yes, this is something that we had previously and somehow we missed this when Codrin was making changes. We want to clean those files due to logic which is inside DevOps. After running tests they are searching for all files *.coverage and merging them. If we keep original files it will be taken and merged 1 more time, which is not needed.

Reply:
Original logic was:

```
                    IList<string> filesToDelete = new List<string>(files.Count) { tempFileName };

                    for (int i = 1; i < files.Count; i++)
                    {
                        cancellationToken.ThrowIfCancellationRequested();
                        progressReporter?.Report(100 * i / files.Count);

                        cancellationToken.ThrowIfCancellationRequested();
                        methodInfo.Invoke(null, new object[] { files[i], outputfileName, tempFileName, true });

                        cancellationToken.ThrowIfCancellationRequested();
                        File.Copy(tempFileName, outputfileName, true);

                        filesToDelete.Add(files[i]);
                    }

                    cancellationToken.ThrowIfCancellationRequested();
                    foreach (string fileName in filesToDelete)
                    {
                        File.Delete(fileName);
                    }
```

I don't know how I missed this during review :)